cartography/intel/aws/ec2.py

```diff
@@ -606,3 +737,129 @@ def sync_load_balancers(session, boto3_session, regions, current_aws_account_id,
+    #             ],
+    #             "State": "available",
+    #             "DhcpOptionsId": "dopt-7a8b9c2d",
+    #             "CidrBlock": "10.0.0.0/16",
+    #             "IsDefault": false
+    #         }
+    #     ]
+    # }
+
+    ingest_vpc = """
+    MERGE (new_vpc:AWSVpc{id: {VpcId}})
+    ON CREATE SET new_vpc.firstseen = timestamp(), new_vpc.vpcid ={VpcId}
+    SET new_vpc.instance_tenancy = {InstanceTenancy},
+    new_vpc.state = {State},
+    new_vpc.is_default = {IsDefault},
+    new_vpc.primary_cidr_block = {PrimaryCIDRBlock},
+    new_vpc.dhcp_options_id = {DhcpOptionsId},
+    new_vpc.lastupdated = {aws_update_tag}
+    WITH new_vpc
+    MATCH (awsAccount:AWSAccount{id: {AWS_ACCOUNT_ID}})
+    MERGE (awsAccount)-[r:RESOURCE]->(new_vpc)
+    ON CREATE SET r.firstseen = timestamp()
+    SET r.lastupdated = {aws_update_tag}"""
+
+    for vpc in data['Vpcs']:
+        vpc_id = vpc["VpcId"]  # fail if not present
+
+        session.run(
+            ingest_vpc,
+            VpcId=vpc_id,
+            InstanceTenancy=vpc.get("InstanceTenancy", None),
+            State=vpc.get("State", None),
+            IsDefault=vpc.get("IsDefault", None),
+            PrimaryCIDRBlock=vpc.get("CidrBlock", None),
+            DhcpOptionsId=vpc.get("DhcpOptionsId", None),
+            AWS_ACCOUNT_ID=current_aws_account_id,
+            aws_update_tag=aws_update_tag)
+
+        load_cidr_association_set(session,
+                                  vpc_id=vpc_id,
+                                  block_type="ipv4",
+                                  vpc_data=vpc,
+                                  aws_update_tag=aws_update_tag)
+
+        load_cidr_association_set(session,
+                                  vpc_id=vpc_id,
+                                  block_type="ipv6",
+                                  vpc_data=vpc,
+                                  aws_update_tag=aws_update_tag)
+
+
+def _get_cidr_association_statement(block_type):
+    ingest_cidr = """
+    MATCH (vpc:AWSVpc{id: {VpcId}})
+    WITH vpc
+    UNWIND {CidrBlock} as block_data
+        MERGE (new_block:#BLOCK_TYPE#{id: {VpcId} + '|' + block_data.#BLOCK_CIDR#})
+        ON CREATE SET new_block.firstseen = timestamp()
+        SET new_block.association_id = block_data.AssociationId,
+        new_block.cidr_block = block_data.#BLOCK_CIDR#,
+        new_block.block_state = block_data.#STATE_NAME#.State,
+        new_block.block_state_message = block_data.#STATE_NAME#.StatusMessage,
+        new_block.lastupdated = {aws_update_tag}
+        WITH vpc, new_block
+        MERGE (vpc)-[r:BLOCK_ASSOCIATION]->(new_block)
+        ON CREATE SET r.firstseen = timestamp()
+        SET r.lastupdated = {aws_update_tag}"""
+
+    BLOCK_CIDR = "CidrBlock"
+    STATE_NAME = "CidrBlockState"
+
+    # base label type. We add the AWS ipv4 or 6 depending on block type
+    BLOCK_TYPE = "AWSCidrBlock"
+    
+    if block_type == "ipv6":
+        BLOCK_CIDR = "Ipv6" + BLOCK_CIDR
```

Comment:
If in the future we need to string search for `Ipv6CidrBlock`, we wouldn't be able to easily find it in the code.  To address this we can add a comment like `# :Ipv6Cidrblock defined here` or something similar.

Reply:
The new nodes are discoverable with str search. The difficulty with the STS_ASSUME role was that the full name of the relationship was a string concat of STS_ASSUME +#Access# 

BLOCK_TYPE = "AWSCidrBlock"
BLOCK_TYPE = BLOCK_TYPE + ":AWSIpv6CidrBlock"
BLOCK_TYPE = BLOCK_TYPE + ":AWSIpv4CidrBlock"